libs/execution_context/ebpf_program.c

```diff
@@ -597,15 +599,7 @@ _ebpf_program_load_machine_code(
     uint8_t* local_machine_code = NULL;
     ebpf_memory_descriptor_t* local_code_memory_descriptor = NULL;
 
-    if (program->parameters.code_type != EBPF_CODE_JIT && program->parameters.code_type != EBPF_CODE_NATIVE) {
-        EBPF_LOG_MESSAGE_UINT64(
-            EBPF_TRACELOG_LEVEL_ERROR,
-            EBPF_TRACELOG_KEYWORD_PROGRAM,
-            "_ebpf_program_load_machine_code program->parameters.code_type must be EBPF_CODE_JIT or EBPF_CODE_NATIVE",
-            program->parameters.code_type);
-        return_value = EBPF_INVALID_ARGUMENT;
-        goto Done;
-    }
+    ebpf_assert(program->parameters.code_type == EBPF_CODE_JIT || program->parameters.code_type == EBPF_CODE_NATIVE);
```

Comment:
This function is called in context of `EBPF_OPERATION_LOAD_CODE` ioctl and `code_type` is passed from user mode.

I think we should not have an assert here an instead have a check for the code type.

Reply:
Line 765 already checks this and only calls this function for these two types, so this assert is safe.